server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferWithLinkedList.java

```diff
@@ -39,11 +41,11 @@ public class ShuffleBufferWithLinkedList extends AbstractShuffleBuffer {
   // blocks will be added to inFlushBlockMap as <eventId, blocks> pair
   // it will be removed after flush to storage
   // the strategy ensure that shuffle is in memory or storage
-  private List<ShufflePartitionedBlock> blocks;
-  private Map<Long, List<ShufflePartitionedBlock>> inFlushBlockMap;
+  private Set<ShufflePartitionedBlock> blocks;
```

Comment:
@zhengchenyu @lwllvyb Do you think this could cause memory leak?

Reply:
> @zhengchenyu @lwllvyb Do you think this could cause memory leak?

I think it's possible

Reply:
It's ok to me.